### ISSUE TITLE
Code2flow pytorch train.py, but results is not good

### ISSUE THREAD

**Issue by PinkR1ver** [scottrogowski/code2flow#48]
![out](https://user-images.githubusercontent.com/50662650/153699413-71e7ef24-ecfd-41f0-a10f-420f74b6b438.png)

**Comment by PinkR1ver** [comment]
Print Message:

Code2Flow: Found 1 files from sources argument.
Code2Flow: Implicitly detected language as 'py'.
Code2Flow: Processing 1 source file(s).
Code2Flow:   .\train.py
Code2Flow: Found groups ['File: train'].
Code2Flow: Found nodes ['(global)'].
Code2Flow: Found calls ['DataLoader()', 'UNKNOWN_VAR.to()', 'UNet()', 'eval()', 'f.close()', 'f.read()', 'f.write()', 'image.to()', 'int()', 'len()', 'lossFunction()', 'mask.to()', 'net()', 'net.load_state_dict()', 'net.parameters()', 'net.state_dict()', 'nn.BCELoss()', 'np.array()', 'open()', 'opt.step()', 'opt.zero_grad()', 'optim.Adam()', 'os.dirname()', 'os.join()', 'os.mkdir()', 'print()', 'testLoss.item()', 'testLossList.averageList_plot()', 'testLossList.averageList_pushback()', 'testLossList.averageList_write_into_log()', 'testLossList.clear_averageList()', 'testLossList.clear_list()', 'testLossList.list_plot()', 'testLossList.list_pushback()', 'testLossList.list_write_into_log()', 'testSensitivityList.averageList_plot()', 'testSensitivityList.averageList_pushback()', 'testSensitivityList.averageList_write_into_log()', 'testSensitivityList.clear_averageList()', 'testSensitivityList.clear_list()', 'testSensitivityList.list_plot()', 'testSensitivityList.list_pushback()', 'testSensitivityList.list_write_into_log()', 'testSpecificityList.averageList_plot()', 'testSpecificityList.averageList_pushback()', 'testSpecificityList.averageList_write_into_log()', 'testSpecificityList.clear_averageList()', 'testSpecificityList.clear_list()', 'testSpecificityList.list_plot()', 'testSpecificityList.list_pushback()', 'testSpecificityList.list_write_into_log()', 'torch.load()', 'torch.random_split()', 'torch.save()', 'torch.stack()', 'torchvision.save_image()', 'trainHelper.sensitivity_and_specificity_calculation()', 'trainHelper.trainHelper()', 'trainLoss.backward()', 'trainLoss.item()', 'trainLossList.averageList_plot()', 'trainLossList.averageList_pushback()', 'trainLossList.averageList_write_into_log()', 'trainLossList.clear_averageList()', 'trainLossList.clear_list()', 'trainLossList.list_plot()', 'trainLossList.list_pushback()', 'trainLossList.list_write_into_log()', 'trainSensitivityList.averageList_plot()', 'trainSensitivityList.averageList_pushback()', 'trainSensitivityList.averageList_write_into_log()', 'trainSensitivityList.clear_averageList()', 'trainSensitivityList.clear_list()', 'trainSensitivityList.list_plot()', 'trainSensitivityList.list_pushback()', 'trainSensitivityList.list_write_into_log()', 'trainSpecificityList.averageList_plot()', 'trainSpecificityList.averageList_pushback()', 'trainSpecificityList.averageList_write_into_log()', 'trainSpecificityList.clear_averageList()', 'trainSpecificityList.clear_list()', 'trainSpecificityList.list_plot()', 'trainSpecificityList.list_pushback()', 'trainSpecificityList.list_write_into_log()'].  
Code2Flow: Found variables ['*->UNKNOWN_MODULE', 'DataLoader->UNKNOWN_MODULE', 'basePath-><Call owner_token=os token=dirname>', 'dataPath-><Call owner_token=os token=join>', 'epoch-><Call owner_token=None token=int>', 'epoches-><Call owner_token=None token=int>', 'f-><Call owner_token=None token=open>', 'fullTrainDataset-><Call owner_token=None token=eval>', 'lossFunction-><Call owner_token=nn token=BCELoss>', 'modelPath-><Call owner_token=os token=join>', 'net-><Call owner_token=UNKNOWN_VAR token=to>', 'nn->UNKNOWN_MODULE', 'np->UNKNOWN_MODULE', 'opt-><Call owner_token=optim token=Adam>', 'optim->UNKNOWN_MODULE', 'os->UNKNOWN_MODULE', 'outImage-><Call owner_token=None token=net>', 'predictPath-><Call owner_token=os token=join>', 
'savePath-><Call owner_token=os token=join>', 'sys->UNKNOWN_MODULE', 'testImage-><Call owner_token=torch token=stack>', 'testLoader-><Call owner_token=None token=DataLoader>', 'testLoss-><Call owner_token=None token=lossFunction>', 'testLossList-><Call owner_token=trainHelper token=trainHelper>', 'testSensitivityList-><Call owner_token=trainHelper token=trainHelper>', 'testSpecificityList-><Call owner_token=trainHelper token=trainHelper>', 
'torch->UNKNOWN_MODULE', 'torchvision->UNKNOWN_MODULE', 'traceback->UNKNOWN_MODULE', 'trainHelper->UNKNOWN_MODULE', 'trainLoader-><Call owner_token=None token=DataLoader>', 'trainLoss-><Call owner_token=None token=lossFunction>', 'trainLossList-><Call owner_token=trainHelper token=trainHelper>', 'trainSensitivityList-><Call owner_token=trainHelper token=trainHelper>', 'trainSize-><Call owner_token=None token=int>', 'trainSpecificityList-><Call owner_token=trainHelper token=trainHelper>', 'weightPath-><Call owner_token=os token=join>'].
Code2Flow: Generating output file...
Code2Flow: Wrote output file 'out.gv' with 0 nodes and 0 edges.
Code2Flow: For better machine readability, you can also try outputting in 
a json format.
Code2Flow: Code2flow finished processing in 0.02 seconds.
Code2Flow: Running graphviz to make the image...
Code2Flow: Graphviz finished in 0.13 seconds.
Code2Flow: Completed your flowchart! To see it, open 'out.png'.

**Comment by scottrogowski** [comment]
Hi @PinkR1ver

Based on:

    Code2Flow: Processing 1 source file(s).
    Code2Flow: .\train.py

My guess is that the single file you are running against, train.py, is a dummy file without any function calls. It looks like there are a lot of "train.py" files in pytorch so I'm not sure which one you are pointed against.

You might try running code2flow against the pytorch directory more like 
    
    code2flow torch/subdirectory/with/many/source/files

One real problem is the fact that it is a little unclear from the output what is happening. I'll fix that soon. Closing this issue for now. Feel free to reopen if I missed something

**Comment by PinkR1ver** [comment]
My expression problem.
I use code2flow to my train.py which uses PyTorch. Here is the link about the train.py:
https://github.com/PinkR1ver/Radiogenemics--on-Ivy-Gap/blob/master/train.py
I am not very sure about this file don't call functions. I guess it has lots of functions to call? 
Thanks for replying!

**Comment by scottrogowski** [comment]
Sorry, missed your reply and coming back to this issue now. Looking at your file, there aren't actually function definitions in it. You would need to include the files you import to get code2flow to work. Working on the README now...
